packages/hdwallet-keepkey/src/eos.ts

```diff
@@ -0,0 +1,204 @@
+import * as Core from '@shapeshiftoss/hdwallet-core'
+
+import { KeepKeyTransport } from './transport'
+
+import {
+  EosGetPublicKey,
+  EosPublicKey,
+  EosPublicKeyKindMap,
+  EosSignTx,
+  EosTxHeader,
+  EosTxActionRequest,
+  EosTxActionAck,
+  EosActionCommon,
+  EosPermissionLevel,
+  EosAsset,
+  EosActionTransfer,
+  EosSignedTx,
+} from '@keepkey/device-protocol/lib/messages-eos_pb'
+import {
+  MessageType
+} from '@keepkey/device-protocol/lib/messages_pb'
+
+import { cloneDeep } from 'lodash'
+
+var Long = require("long")
+
+export function eosGetAccountPaths (msg: Core.EosGetAccountPaths): Array<Core.EosAccountPath> {
+  return [{
+    addressNList: [ 0x80000000 + 44, 0x80000000 + Core.slip44ByCoin('Eos'), 0x80000000 + msg.accountIdx, 0, 0 ]
+  }]
+}
+
+
+function charToSymbol(c:string) : number {
+  if (c.charCodeAt(0) >= "a".charCodeAt(0) && c.charCodeAt(0) <= "z".charCodeAt(0)) {
+    return ((c.charCodeAt(0) - "a".charCodeAt(0)) + 6)
+  }
+  if (c.charCodeAt(0) >= "1".charCodeAt(0) && c.charCodeAt(0) <= "5".charCodeAt(0)) {
+    return ((c.charCodeAt(0) - "1".charCodeAt(0)) + 1)
+  }
+  return 0
+}
+
+function nameToNumber(name:string) : string {
+  var Long = require("long")
+  let value = new Long(0, 0, true)
+  let c = new Long(0, 0, true)
+
+  for (let i=0; i<13; i++) {
+    if (i < name.length && i < 13) {
+        c = Long.fromNumber(charToSymbol(name[i]), true)
+    }
+
+    if (i < 12) {
+        c = c.and(Long.fromString("1f", true, 16))
+        c = c.shl(64 - 5 * (i + 1))
+
+    } else { // last iteration
+        c = c.and(Long.fromString("0f", true, 16))
+    }    
+      value = value.or(c)
+  }
+  return value.toString()
+}
+
+function assetToNumber(asset:string): [number, number] {
```

Comment:
whats this do?

Reply:
Same as the nameToNumber(), converts the asset name to a number for signing. Derived from the trezor reference for testing.